lib/iris/experimental/regrid.py

```diff
@@ -1105,9 +1252,25 @@ def __call__(self, src):
                              'source grid as this regridder.')
 
         # Call the regridder function.
-        res = regrid_weighted_curvilinear_to_rectilinear(src, self.weights,
-                                                         self._target_cube)
-        return res
+        # This includes repeating over any non-XY dimensions, because the
+        # underlying routine does not support this.
+        # FOR NOW: we will use cube.slices and merge to achieve this,
+        # though that is not a terribly efficient method ...
```

Comment:
But it will do for now :smile: We could always look into duplicating the behaviour of the "big" regridders for this functionality at a later date.